pkgs/dart_model/lib/src/dart_model.dart

```diff
@@ -28,6 +30,11 @@ extension ParentInterface on Member {
 }
 
 extension ModelExtension on Model {
+  /// An identity hash for `this`, used for comparing query results.
+  ///
+  /// TODO: A faster/better implementation?
+  int get identityHash => const DeepCollectionEquality.unordered().hash(node);
```

Comment:
Fwiw, it looks like the implementation of this does not use `entries`, but instead uses `keys` and then reads the value of the key, so this might explain some of the slowness.

Reply:
Ok, yeah wow I hacked a version of package:collection to use `entries` and it is at least twice as fast, and now faster overall with the cached results than without. I might look at  upstreaming that, and if not then we can have our own implementation.

Reply:
Ordered comparison might be faster, too? If the ordering is sufficiently stable...

Reply:
yeah about 10%, switching to that.